File: PrimeNumber.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

readline.question("Digite um número para saber se é número primo: ", number => {  
  let numero = Number(number);
  
  verificarNumeroPrimo(numero);
  readline.close();
});

function verificarNumeroPrimo(isNumber){
  let index, model, flag = 0;
  if (typeof isNumber === 'number'){
      model = isNumber / 2;
      if (isNumber == 0 || isNumber == 1){
        console.log(isNumber + " is not Prime number ");
      } else {
          for(index = 2; index <= model; index++) {
              if(isNumber % index == 0){
                  console.log(isNumber + " is not Prime number ");
                  flag = 1;
                  break;
              }
              if (flag == 0){
                  console.log(isNumber + " is Prime number ");
                  break;
              }
          }
      }
  } else {
      console.log("Impossivel calcular");
  }   
}

File: PrimeNumber.java
import java.util.Scanner;
public class PrimeNumber{
    public static void main(String[] args) {
        int numero;
        int index, model, flag = 0;
        
        Scanner scan = new Scanner(System.in);
        System.out.println("Digite um número para saber se é número primo: ");
        numero = scan.nextInt();
        
        model = numero/2;

        if(numero == 0 || numero == 1){
            System.out.println(numero + " is not Prime number");
        } else {
            for (index = 2; index <= model; index++) {
                if(numero % index == 0){
                    System.out.println(numero +" is not prime number");
                    flag = 1;
                    break;
                }
                if (flag == 0) {
                    System.out.println(numero+" is prime number");
                    break;
                }

            }
        }

    }
}